### PR TITLE
[libsquoosh] Add thread support to AVIF in Node

### PR DESCRIPTION
**Pull request by surma**

### CODE REVIEW

File: libsquoosh/src/codecs.ts [hunk 1]
```diff
@@ -1,5 +1,13 @@
 import { promises as fsp } from 'fs';
 import { instantiateEmscriptenWasm, pathify } from './emscripten-utils.js';
+import { threads } from 'wasm-feature-detect';
+import { cpus } from 'os';
+
+// We use `navigator.hardwareConcurrency` for Emscripten’s pthread pool size.
+// This is the only workaround I can get working without crying.
```
Comment (nucliweb) [inline libsquoosh/src/codecs.ts:7]: To maintain consistency with the previous sentence
```suggestion
// This is the only workaround we can get working without crying.
```

File: libsquoosh/src/emscripten-utils.ts [hunk 2]
```diff
@@ -10,10 +10,14 @@ export function pathify(path: string): string {
 export function instantiateEmscriptenWasm<T extends EmscriptenWasm.Module>(
   factory: EmscriptenWasm.ModuleFactory<T>,
   path: string,
+  workerWasm: string = '',
```
Comment (RReverser) [inline libsquoosh/src/emscripten-utils.ts]: As far as I can tell, this should be called `workerJS` not `workerWasm`.

File: libsquoosh/src/emscripten-utils.ts [hunk 3]
```diff
@@ -10,10 +10,14 @@ export function pathify(path: string): string {
...
 ): Promise<T> {
   return factory({
-    locateFile() {
-      return pathify(path);
+    locateFile(requestPath) {
+      if (requestPath.endsWith('.wasm')) return pathify(path);
```
Comment (RReverser) [inline libsquoosh/src/emscripten-utils.ts:21]: I'll admit I don't understand this logic / why we need different logic for different types of files here. Maybe a comment would be helpful?

Comment (surma) [reply libsquoosh/src/emscripten-utils.ts:21]: The build system for libSquoosh doesn’t understand `new URL(..., import.meta.url)` for assets yet, so we need to manually inject the updated paths here.

File: libsquoosh/src/emscripten-utils.ts (hunk 2 above)
Comment (surma) [reply libsquoosh/src/emscripten-utils.ts]: Good point

File: libsquoosh/src/emscripten-utils.ts (hunk 3 above)
Comment (surma) [reply libsquoosh/src/emscripten-utils.ts:21]: I’ll add a comment

**Pull request merged**